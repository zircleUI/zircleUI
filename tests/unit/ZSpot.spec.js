import { shallowMount, createLocalVue } from '@vue/test-utils'
import zspot from '@/components/z-spot.vue'
import zslider from '@/components/child-components/z-slider.vue'
import zircle from '@/index'

const localVue = createLocalVue()
localVue.use(zircle)
const wrapper = shallowMount(zspot, {
  propsData: {
    slider: true,
    progress: 80,
    imagePath: './images/test.png',
    label: 'Test label prop'
  },
  provide: function () {
    return {
      view: 'test'
    }
  },
  computed: {
    position: function () {
      return { X: 0, Y: 0, scale: 1, Xi: 0, Yi: 0, scalei: 1 }
    }
  },
  slots: {
    default: '<p> lorem text </p>',
    media: '<div>youtube video</div>',
    imagePath: './images/dummy.png',
    label: 'Test label slot',
    extension: '<div></div>'
  },
  stubs: {
    'z-slider': zslider
  },
  localVue
})
describe('z-spot.vue', () => {
  it('Renders props when passed', () => {
    expect(wrapper.vm.imagePath).toEqual('./images/test.png')
    expect(wrapper.vm.label).toEqual('Test label prop')
  })

  it('Has z-slider component and progress 80%', () => {
    expect(wrapper.vm.progress).toEqual(80)
    expect(wrapper.findComponent(zslider).exists()).toBe(true)
  })
  it('Has the expected html structure', () => {
    // Note: If props.imagesrc and or props.label are present, slots.imagesrc and slots.label wont be rendered
    expect(wrapper.element).toMatchSnapshot()
  })
})
