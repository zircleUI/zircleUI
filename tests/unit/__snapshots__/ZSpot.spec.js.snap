// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`z-spot.vue Has the expected html structure 1`] = `
<div
  class="z-shape is-extension z-spot primary"
  role=""
  style="transform: translate3d(0px, 0px, 0px);"
>
  <div
    class="z-outer-spot"
  />
   
  <div
    class="z-pulse"
  />
   
  <!---->
   
  <svg
    class="z-slider"
    viewBox="0 0 100 100"
    xmlns="http://www.w3.org/2000/svg"
  >
    <circle
      cx="50"
      cy="50"
      r="51"
      style="transform-origin: 50% 50%; transform: rotate(-90deg); stroke-dasharray: 320.4424506661589; stroke-dashoffset: 64.08849013323174; stroke-width: 8;"
    />
  </svg>
   
  <div
    class="z-label bottom"
  >
    <div
      class="inside"
    >
      
      Test label prop 
      <!---->
    </div>
  </div>
   
  <div
    class="z-content"
  >
    <img
      alt="content custom image"
      height="100%"
      src="./images/test.png"
      width="100%"
    />
     
    <!---->
  </div>
   
  <div
    class="z-content"
    style="z-index: 10;"
  >
    <span
      class="overflow"
    >
      <span>
        
      </span>
       
      <p>
         lorem text 
      </p>
    </span>
  </div>
   
  <div />
</div>
`;
